### ISSUE TITLE
Optimize RPC headers

### ISSUE THREAD

**Issue by TheYellowArchitect** [godotengine/godot-proposals#10885]
### Describe the project you are working on

Online topdown shooter (think Synthetik) using rollback (netfox plugin)

### Describe the problem or limitation you are having in your project

A rollback user opened an issue https://github.com/foxssake/netfox/issues/290 because he noticed the following quote from 
 https://github.com/jonathaneeckhout/jdungeon/blob/main/documentation/networkingsync.md

>You may also note that argument names for these functions are a single letter. This is intentional due to Godot needing to send argument names when performing RPCs.

And he is right. RPCs passing parameters as strings is wildly inefficient, especially using rollback where every byte matters (since multiple states can be sent at once)

**tl;dr: To improve your bandwidth over RPCs, you gotta use single-letter parameter names, instead of descriptive parameter names.**

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Send the parameters normally like every language does, where arguments go by order, so no strings are passed.
I honestly doubt ENET passes parameters like this. If I am proven wrong, or this issue is false, please tell me so I close it.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Use normal parameter names in your RPC functions, instead of trying to win bandwidth by making your code unreadable.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Yes, you just use single-letter variable names in the parameters of every RPC...

### Is there a reason why this should be core and not an add-on in the asset library?

Its an ENET binding bug. I cannot see how this could be core or an addon.

**Comment by Faless** [comment]
> And he is right.

No, this is completely false.

Godot **does not** send the arguments names, **nor the function name**.

The RPC sends:

- The Node Network ID **OR** the node path (only if the node path has not been already exchanged)
- The function ID (reason why RPCs must match between server and client)
- The argument count
- The argument values

**Comment by TheYellowArchitect** [comment]
Thank you for clearing out the confusion, so this claim doesn't spread further :+1:
